### DISCUSSION TITLE
How to integrate plausible analytics?

### DISCUSSION THREAD

**Discussion by Always-Self-Hosted** [dani-garcia/vaultwarden#4329] [Q&A] [answered]
Hi all,

I am interested in using my own plausible analytics with my self hosted vaultwarden. In other services I've done this with, there has been support for overwriting the html templates with your own, or some super nice examples (like [gitea](https://docs.gitea.com/administration/customizing-gitea#other-additions-to-the-page)) where they have dedicated files specific for just adding in things like additional scripts to the end of the head block in the main templates. 

I'm looking around the wiki and am wondering if this is possible with vaultwarden? there does seem to be handlebar templates in the repo, but i can't see an index.html, infact i cant see any "normal" pages, only those under the /admin path? And there is no specific documentation on adding things to the html templates (unless i've missed something in the wiki)

For context for those who are not familiar with integrating plausible, it requires adding the .js file in the head block, and some inline functions in the body block to link pages to "events" (such as create account, or login ect).

Any help on this would be greatly appreciated.

**Comment by BlackDex** [comment, marked as answer]
We do not maintain the web-vault. It's a compiled version from Bitwarden with some modifications from our side to make it work with the Vaultwarden backend 

You probably need to build the web-vault you're self if you want to do this. It maybe modifying dinner files inn the web-vault might work.

See https://github.com/dani-garcia/bw_web_builds for our patches, build scripts, and the final released files.

**Reply by Always-Self-Hosted** [reply]
Thanks for the response. This definitely seems like a bit too much work, but great to know this is the answer to the question.

**Comment by RealOrangeOne** [comment]
Adding analytics to your password manager is probably not a great idea from a privacy and security perspective. Loading extra JS into the browser is another attack vector - anyone with access to update Plausible can steal credentials. If you just want to be able to see who has done what, check out Bitwarden's built-in [audit log](https://bitwarden.com/help/event-logs/) (as that'll work for non-browser clients too.

If you do need to do it, you could also do it with a reverse proxy, and inject the `script` tag just after the `</body>`. Although you'd also likely need to add your plausible domain to the CSP header so the browser can communicate with it.

But, again, and I can't stress this enough, _please_ don't.

**Reply by Always-Self-Hosted** [reply]
I appreciate you taking the time to pass on the information, this is a realm I don't have much experience in at all. As you guessed, I mostly want eyes on whats happening to this instance without going through the logs all the time. This is a public instance with sign ups enabled for my family / friends to use without having to get an invite link from me, but i've already had to block port 22 with ufw because this box was getting smashed with login attempts by bots over ssh, and i really just want to keep tabs on the traffic going to it. 

I will look into your suggestion about the audit log and see if thats any easier than literally cat'ing the log files on the box.

**Reply by Always-Self-Hosted** [reply]
Oh it looks like that feature is specific to orgs, which i don't use. thats a shame. But still, thank you for the advice.
